src/main/java/org/marketdesignresearch/mechlib/utils/CPLEXUtils.java

```diff
@@ -19,15 +18,55 @@
 
     private final Map<SolveParam, Object> solveParamMap = Maps.newHashMap();
 
+    /**
+     * Solves the program.
+     * It applies the solve parameters that are specified in {@link #solveParamMap} only if they are not already
+     * specified in the program itself.
+     * Careful: JOpt has a default parameter logic as well, so often the program already includes JOpt-specific default
+     * parameters, which will not be overwritten here. To be sure, you can call {@link IMIP#clearSolveParams()} before
+     * setting your own parameters
+     * @param program The program to be solved
+     * @return The result object
+     */
     public IMIPResult solve(IMIP program) {
-        solveParamMap.forEach(program::setSolveParam);
+        solveParamMap.forEach((param , value) -> {
+            if (!program.isSolveParamSpecified(param)) {
+                program.setSolveParam(param, value);
+            }
+        });
         return solver.solve(program);
     }
 
+    /**
+     * You can add additional (or override existing) default parameters that are then applied to any future program
+     * that does not define these solve parameters by itself already.
+     * @param param The SolveParam to be specified
+     * @param value The value that should be set for the SolveParam
+     */
+    public void setSolveParam(SolveParam param, Object value) {
+        solveParamMap.put(param, value);
+    }
+
+
+    /**
+     *  A helper function to initialize the solver with some reasonable default parameters.
+     *  Has to be actively called by the user.
+     */
     public void initializeSolveParams() {
+        solveParamMap.clear();
+        solveParamMap.put(SolveParam.DISPLAY_OUTPUT, log.isDebugEnabled());
+        solveParamMap.put(SolveParam.THREADS, 1);
+        solveParamMap.put(SolveParam.TIME_LIMIT, (double) TimeUnit.SECONDS.convert(1, TimeUnit.HOURS));
```

Comment:
I agree that the DISPLAY_OUTPUT parameter would be a good choice for a "default" parameter. However the other two parameters I would not suggest to set as reasonalbe defaults. Setting threads to 1 is not a good choice if you have more cores and setting a time limit may prevent finding an optimal solution, even though you are really interested in an optimal solution.